src/Runner.Worker/ActionManager.cs

```diff
@@ -1068,8 +928,21 @@ private ActionContainer PrepareRepositoryActionAsync(IExecutionContext execution
                 }
                 else if (actionDefinitionData.Execution.ExecutionType == ActionExecutionType.Composite)
                 {
-                    Trace.Info($"Action composite: {(actionDefinitionData.Execution as CompositeActionExecutionData).Steps}, no more preparation.");
-                    return null;
+                    // TODO: we need to generate unique Id's for composite steps
+                    Trace.Info($"Loading Composite steps");
+                    var compositeAction = actionDefinitionData.Execution as CompositeActionExecutionData;
+                    setupInfo.Steps = compositeAction.Steps;
+
+                    foreach (var step in compositeAction.Steps)
+                    {
+                        step.Id = Guid.NewGuid();
+                        if (string.IsNullOrEmpty(executionContext.Global.Variables.Get("DistributedTask.EnableCompositeActions")) && step.Reference.Type != Pipelines.ActionSourceType.Script)
+                        {
+                            throw new Exception("Composite Actions with the `uses:` keyword are not supported on this server. Please update your GHES or GHAE instance.");
```

Comment:
Should we just say `uses` not supported, as we would if they had an old runner that didn't support the keyword? This error message doesn't make sense for dotcom while we're in the process of rolling out the feature flag.